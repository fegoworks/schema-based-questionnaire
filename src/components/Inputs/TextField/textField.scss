.TextField {
  margin-bottom: 0.6rem;
  display: flex;
  flex-direction: column;
}

.TextField input {
  padding: 0.6rem 0.6rem;
  font-size: 1em;
  font-weight: 500;
  transition: border-color 0.25s;
  width: 15rem;
}

.TextField .error {
  color: #d44;
  margin: 0;
}

.TextField input:focus-visible,
.TextField input:focus {
  border-radius: 5px;
  outline: 4px auto -webkit-focus-ring-color;
  outline-color: #ddd;
}

@media (prefers-color-scheme: light) {
  .TextField input:focus-visible,
  .TextField input:focus {
    outline-color: #333;
  }

  .TextField input:hover {
    outline: 4px auto -webkit-focus-ring-color;
    outline-color: #999;
  }
}
