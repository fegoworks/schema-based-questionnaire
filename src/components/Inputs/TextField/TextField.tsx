import { FieldProps } from "../../formField";
import "./textField.scss";
import { FunctionComponent } from "react";


const TextField: FunctionComponent<FieldProps> = ({
  error,
  name,
  onBlur,
  onChange,
  touched,
  type = "text",
  value,
  disabled
}) => {
  return (
    <div className="TextField">
      <input
        role="textbox"
        id={name}
        name={name}
        onBlur={onBlur}
        onChange={onChange}
        type={type}
        value={value}
        disabled={disabled}
      />
      {error && touched && <p className="error">{error}</p>}
    </div>
  );
};

export default TextField;
