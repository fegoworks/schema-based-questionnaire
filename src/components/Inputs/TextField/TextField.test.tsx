import { fireEvent, render } from "@testing-library/react";
import { describe, expect } from "vitest";
import TextField from "./TextField";

const defaultProps = {
  name: "FirstName",
  disabled: false,
};

describe("Component - TextField", () => {
  it("should render TextField component", () => {
    const { container } = render(<TextField {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("should allow text input", () => {
    const { container } = render(<TextField {...defaultProps} />);
    const input = container.getElementsByTagName("input")[0];

    fireEvent.change(input, { target: { value: "Oghenefego" } });
    expect(input).toHaveDisplayValue("Oghenefego");
  });

  it("should show an error message if validation fails", () => {
    const { getByText } = render(
      <TextField {...defaultProps} error={"Some error"} touched={true} />
    );
    expect(getByText("Some error")).toBeTruthy();
  });
});
