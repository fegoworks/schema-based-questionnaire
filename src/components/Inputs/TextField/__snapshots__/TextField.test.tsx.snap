// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component - TextField > should render TextField component 1`] = `
<div>
  <div
    class="TextField"
  >
    <input
      id="FirstName"
      name="FirstName"
      role="textbox"
      type="text"
      value=""
    />
  </div>
</div>
`;
